### DISCUSSION TITLE
some curious problems, bug or my system

### DISCUSSION THREAD

**Discussion by guenther-alka** [openzfsonwindows/openzfs#371] [General] [answered]
I have bunch of problems

It started when I tried to access snapshots in a folder tank/data/.zfs
I could see the snapshot folder but Windows explorer hangs when I tried to open

I could not kill explorer nor could I reboot (system hangs similar to a pool suspended).
A hard reset was required

The system came up with a bsod, a second boot worked

As I had installed Open-ZFS 2.2.3 rc2 dirty I installed 2.2.3 rc2 newest.
Installer crashed at the end of setup

I then decided to deinstall Open-ZFS first.
This failed with a "corrupted file" message.

I checked device manager for options to deinstall driver
There I saw 4 OpenZFS drivers in storagevolumes 
I could deinstall one with a crash on the next one.
(checked again after recovery, there are now two)

A side effect: I could no longer start "Terminal as admin"
I could fix this with an update of Terminal via MS store.

I retried to reinstall Open-ZFS newest.
Again the system crashed with a bsod message "bad system config info"
Only way to recover was a go back to last recovery point. 

This is newest Win 11 Pro
History: Update problems started with 2.2.3 rc1 with a crash due the cpuid problem

**Comment by lundman** [comment]
Sorry you had many issues, sounds like I should run the tests for .zfs access for one, as it appears to not play nicely with explorer. As for removing the driver, you can just delete the OpenZFS.sys file from Windows/system32/drivers - this can be done from the boot recovery console. it's the only file it installs.

I thought I had deleted the cpuid buggy version, I'll go check that.

**Comment by guenther-alka** [comment]
Since that buggy rc1 version I have problems with updates to newer releases or to deinstall.

ps 
Just tried to update to rc3 from rc2 dirty. Prior the end of the update Windows crashed. On reboot it crashed again. Second boot worked. I checked System > Settings  > Apps then. State is still rc2 dirty. 

btw
I now have again 3 x OpenZFS controller in device manager. Not sure if it should be 1,2 or more.

**Comment by lundman** [comment]
Remove OpenZFS.sys, just delete the file, reboot. then use device manager to clean up any installs. this isnt needed, but if you want it to look clean, you might as well.
Then install rc3.

**Comment by guenther-alka** [comment, marked as answer]
I was able to remove:

- rename OpenZFS.sys to OpenZFS.sys.xx (delete not possible but rename)
- reboot
- remove the three OpenZFS controllers that a found in device > manager >storagevolumes
- reboot
- tried to deinstall rc2 dirty: failed (message file corrupted)
-install rc3: ok
-tried deinstall : ok
- reinstalled rc3: ok
- opened device manager: crash with irql not less or equal in netio.sys

It seeems this is another problem propably introduced by the various crashes.

**Comment by sskras** [comment]
@guenther-alka, which Windows build is that?

**Reply by lundman** [reply]
Ah yes, if it's loaded you probably can't delete it while booted, but rename is a good way around it.

**Reply by guenther-alka** [reply]
Windows 11 Pro german, 23H2, up to date.

btw
The problem with snapshot access via Explorer remains ex 
tank/data/.zfs

I cannot open, Explorer hangs, not even a kill of file explorer was possible

![image](https://github.com/openzfsonwindows/openzfs/assets/27874553/814f50f3-e0aa-4ddd-9673-6228f3c8a408)

Reboot needed to kill

**Reply by lundman** [reply]
Yeah .zfs works fine here, so it could be related to changes in Win11Pro - like the SMB bug. I am in process of setting up VM

**Comment by datacore-rm** [comment]
We had fixed similar issue of multiple controllers added in device manager during upgrade install.
https://github.com/DataCoreSoftware/openzfs/commit/d4007a884e4a2e086ab914759495753769974aac

**Comment by lundman** [comment]
Ah neat - thanks. I shall cherry-pick

**Comment by datacore-rm** [comment]
@lundman There is another ZfsInstaller issue during driver uninstall. It does not delete the driver package from the Driver Store. 
FYI, it is fixed in the below PR. You can please cherry-pick if you think so. Thanks.

(#define DRIVER_SYS_FILE to be changed accordingly) 
 https://github.com/DataCoreSoftware/openzfs/pull/97

**Reply by mcladams** [reply]
https://github.com/openzfsonwindows/openzfs/issues/430#issuecomment-2629500497
Most of the symptoms described were also found over here. The subject issue is closed but some of the discussions about removing drivers, (re)installation and multiple copies the same. I solved using Dism from a different install of windows on a different partition on the same machine.  (All of my windows machines have two editions of windows on them for different purposes, but the nicest thing about that is being about to use Dism on the windows that's offline from the running one without the limitations of either the Recovery Environment or the limitation of working on the online image). https://github.com/openzfsonwindows/openzfs/issues/430#issuecomment-2629500497

**Comment by lundman** [comment]
Cheers, appreciate you guys sharing!

Do you hardcore Windows guys know much about pnpmanager to mount volumes? I am redoing that part so mounts are more valid, and work more correctly, but it is being most difficult and chatgpt is barely helping :)

**Comment by guenther-alka** [comment]
I just tried to create an encrypted filesystem (win11, open-zfs rc.6) via command

`zfs create -o utf8only=on -o normalization=formD -o compress=lz4 -o encryption=aes-256-gcm -o keyformat=passphrase -o keylocation=file:///xampp/web-gui/tmp/localhost~127.0.0.1~tank~enc.completekey -o snapdir=hidden -o casesensitivity=insensitive -o nbmand=on -o sharesmb=off -o atime=off tank/enc`

File with key exists but filesystem not created (command runs without an error)
Is this related to current rc6 restrictions or a bug?

**Reply by sskras** [reply]
@guenther-alka if you could put ``` both above and below your command, it would become IMO more readable :)

EDIT: or maybe just prepend and append you command with a single backtick ( ` ).
